tests/unit/testsuite/Bolt/Boltpay/Model/Admin/ExtraConfigTest.php

```diff
@@ -0,0 +1,373 @@
+            ->with($this->stringStartsWith('Invalid JSON for Bolt Extra Options.'));
+
+        $this->currentMock->save();
+    }
+
+    /**
+     * @test
+     * that save method doesn't change value when all parameters are valid
+     *
+     * @covers ::save
+     * @covers ::hasValidBoltPrimaryColor
+     * @covers ::hasValidDatadogKeySeverity
+     * @covers ::hasValidPriceFaultTolerance
+     *
+     * @throws Exception when there is an error saving the extra config to the database
+     */
+    public function save_withValidParameters_savesNewValue()
+    {
+        $value = json_encode(
+            array(
+                'priceFaultTolerance' => 1,
+                'boltPrimaryColor'    => '#ffffff',
+                'datadogKeySeverity'  => Boltpay_DataDog_ErrorTypes::TYPE_ERROR
+            )
+        );
+        $this->currentMock->expects($this->once())->method('getValue')->willReturn($value);
+        $this->currentMock->expects($this->never())->method('setValue');
+        $this->currentMock->save();
+    }
+
+    /**
+     * @test
+     * that save method will save previous value if any of the options is invalid
+     *
+     * @covers ::save
+     * @covers ::hasValidPriceFaultTolerance
+     *
+     * @throws Mage_Core_Exception if unable to stub singleton
+     * @throws Exception when there is an error saving the extra config to the database
+     */
+    public function save_withInvalidOptions_savesOldValue()
+    {
+        $invalidPriceFaultTolerance = -1;
+        $value = json_encode(
+            array(
+                'priceFaultTolerance' => $invalidPriceFaultTolerance,
+                'boltPrimaryColor'    => '##',
+                'datadogKeySeverity'  => '##'
+            )
+        );
+        $oldValue = '{}';
+        $sessionMock = $this->getMockBuilder('Mage_Core_Model_Session_Abstract')
+            ->setMethods(array('addError'))
+            ->getMock();
+
+        Bolt_Boltpay_TestHelper::stubSingleton('core/session', $sessionMock);
+
+        $this->currentMock->expects($this->once())->method('getValue')->willReturn($value);
+        $this->currentMock->expects($this->once())->method('getOldValue')->willReturn($oldValue);
+        $this->currentMock->expects($this->once())->method('setValue')->with($oldValue);
+
+
+        $sessionMock->expects($this->once())->method('addError')
+            ->with(
+                "Invalid value for extra option `priceFaultTolerance`.[" . $invalidPriceFaultTolerance . "]
+                         A valid value must be a positive integer."
+            );
+
+        $this->currentMock->save();
+    }
+
+    /**
+     * @test
+     * that hasValidBoltPrimaryColor returns false for invalid color hex and adds error message to session
+     *
+     * @covers ::hasValidBoltPrimaryColor
+     *
+     * @throws Mage_Core_Exception if unable to stub singleton
+     */
+    public function hasValidBoltPrimaryColor_withInvalidColorHex_returnsFalseAndAddsErrorMessageToSession()
```

Comment:
@mccottry Can you add a new unit test named hasValidBoltPrimaryColor_withValidColorHex_returnsTrue?

Reply:
I can update the test for you.